app/queries/servers/role.ts

```diff
@@ -37,7 +37,10 @@ export const queryRolesByNames = (database: Database, names: string[]) => {
     return database.get<RoleModel>(ROLE).query(Q.where('name', Q.oneOf(names)));
 };
 
-export function observePermissionForChannel(database: Database, channel: ChannelModel, user: UserModel, permission: string, defaultValue: boolean) {
+export function observePermissionForChannel(database: Database, channel: ChannelModel | null | undefined, user: UserModel | undefined, permission: string, defaultValue: boolean) {
```

Comment:
why " channel: ChannelModel **| null | undefined**  " of optional args.... is it because of TS issue ?

Reply:
Short answer... yes TS issue.

Long answer...
There are two things going on here. First of all... `null` is not the same as `undefined`. But if I remember correctly, the watermelonDB relations return the model or null, not undefined. So one of the places this function was being used, the value was of type `Model | null` instead of `Model | undefined`. That is why we are including the null.
Second thing is that you cannot have required arguments after optional ones. I could have moved the channel and user to the end, but seemed less readable. And the reality is that they are not optional arguments, they are required arguments (you always want to pass a value to them) that may have undefined as the value.